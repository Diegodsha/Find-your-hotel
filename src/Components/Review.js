import PropTypes from 'prop-types';
import StyledReview from '../Assets/StyledReview';

const Review = ({ review }) => {
  const date = review.created_at;
  const tIndex = date.indexOf('T');
  const dateToWords = new Date(date.slice(0, tIndex)).toDateString();

  return (
    <StyledReview className="card text-center col-12 col-md-5 col-lg-3 mb-4 justify-content-center">
      <div className="card-body m-0">
        <p className="card-text">{review.title}</p>
        <p className="card-text">{review.body}</p>
        <p className="card-text text-secondary">{dateToWords}</p>
      </div>
    </StyledReview>
  );
};

Review.propTypes = {
  review: PropTypes.shape({
    title: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired,
    created_at: PropTypes.string.isRequired,
  }).isRequired,
};

export default Review;
