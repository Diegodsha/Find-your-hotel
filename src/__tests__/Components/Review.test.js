import renderer from 'react-test-renderer';
import Review from '../../Components/Review';

describe('Review Component', () => {
  const review = {
    title: 'review',
    body: 'body test',
    score: 3,
    created_at: '20-02-20',
  };
  test('Renders correctly component', () => {
    const tree = renderer.create(<Review review={review} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
