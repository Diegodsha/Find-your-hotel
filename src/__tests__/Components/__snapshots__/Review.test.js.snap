// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Component Renders correctly component 1`] = `
<div
  className="sc-bdnxRM dyiXzt card text-center col-12 col-md-5 col-lg-3 mb-4 justify-content-center"
>
  <div
    className="card-body m-0"
  >
    <p
      className="card-text"
    >
      review
    </p>
    <p
      className="card-text"
    >
      body test
    </p>
    <p
      className="card-text text-secondary"
    >
      Invalid Date
    </p>
  </div>
</div>
`;
